clang-3.4 detected unreachable code

FAILED: /usr/bin/clang++   -DAMBER_CTS_VULKAN_HEADER=1 -DAMBER_ENABLE_SHADERC=0 -DAMBER_ENABLE_SPIRV_TOOLS=0 -DAMBER_ENGINE_DAWN=0 -DAMBER_ENGINE_VULKAN=1 -DDE_ASSERT_FAILURE_CALLBACK -DDE_COMPILER=DE_COMPILER_CLANG -DDE_CPU=DE_CPU_X86_64 -DDE_DEBUG -DDE_OS=DE_OS_UNIX -DDE_PTR_SIZE=8 -Wall -Wextra -Wno-long-long -Wshadow -Wundef -Wconversion -Wno-sign-conversion -Werror -Wno-error=unused-function -Wno-error=unused-command-line-argument -m64 -std=c++03 -Wno-delete-non-virtual-dtor -fwrapv -fvisibility=hidden -fvisibility-inlines-hidden -std=c++11 -fvisibility=hidden -Wall -Werror -Weverything -Wextra -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-format-pedantic -Wno-padded -Wno-switch-enum -Wno-unknown-pragmas -Wno-unknown-warning-option -g -fPIC -I../../external/amber/../vulkancts/framework/vulkan -I../../framework/delibs/debase -I../../framework/delibs/decpp -I../../framework/delibs/depool -I../../framework/delibs/dethread -I../../framework/delibs/deutil -I../../framework/delibs/destream -I../../external/zlib/src -I../../external/vkrunner/src -I../../external/spirv-headers/src/include -I../../external/amber/src/include -I../../external/amber/src -I.    -fno-exceptions -fno-rtti -Wno-zero-as-null-pointer-constant -MMD -MT external/amber/src/src/vulkan/CMakeFiles/libamberenginevulkan.dir/pipeline.cc.o -MF "external/amber/src/src/vulkan/CMakeFiles/libamberenginevulkan.dir/pipeline.cc.o.d" -o external/amber/src/src/vulkan/CMakeFiles/libamberenginevulkan.dir/pipeline.cc.o -c ../../external/amber/src/src/vulkan/pipeline.cc

../../external/amber/src/src/vulkan/pipeline.cc:44:17: error: will never be executed [-Werror,-Wunreachable-code]

  return Result("Unknown resource type");

                ^~~~~~~~~~~~~~~~~~~~~~~

--- a/src/vulkan/pipeline.cc
+++ b/src/vulkan/pipeline.cc
@@ -33,14 +33,18 @@ namespace {
 const char* kDefaultEntryPointName = "main";
 
 Result ToVkDescriptorType(DescriptorType type, VkDescriptorType* ret) {
+  Result r = Result("Unknown resource type");
   switch (type) {
     case DescriptorType::kStorageBuffer:
       *ret = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER;
-      return {};
+      r = {};
+      break;
     case DescriptorType::kUniformBuffer:
       *ret = VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER;
-      return {};
+      r = {};
+      break;
   }
+  return r;
 }
 
 }  // namespace

--- a/src/vulkan/pipeline.cc
+++ b/src/vulkan/pipeline.cc
@@ -41,7 +41,6 @@ Result ToVkDescriptorType(DescriptorType type, VkDescriptorType* ret) {
       *ret = VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER;
       return {};
   }
-  return Result("Unknown resource type");
 }
 
 }  // namespace

